samples/python_interop/cirq_samples/custom_gates.ipynb

```diff
@@ -0,0 +1,241 @@
+    "        return 'PM_M','PM_A'\n",
+    "\n",
+    "BP = BellPairGate()\n",
+    "MP = MessagePrepGate()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9e0fcb35",
+   "metadata": {},
+   "source": [
+    "## 3. Teleportation with Custom Gates"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "de07ff35",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def simple_teleportation() -> Sequence[cirq.Operation]:\n",
+    "    alice = cirq.NamedQubit('alice')\n",
+    "    bob = cirq.NamedQubit('bob')\n",
+    "    msg = cirq.NamedQubit('msg')\n",
+    "\n",
+    "    program: Sequence[cirq.Operation] = []\n",
+    "    # 1. Prepare Bell pair between alice and bob\n",
+    "    program.append(BP.on(alice, bob))\n",
+    "    # 2. Prepare message qubit in Rx(0.7)|0>\n",
+    "    program.append(cirq.rx(0.7)(msg))\n",
+    "    # 3. Entangle message with alice (Bell + measurement portion of teleportation)\n",
+    "    program.append(MP.on(msg, alice))\n",
+    "    # 4. Measure message and alice to obtain classical bits b1, b2\n",
+    "    program.append(cirq.measure(msg, key='b1'))\n",
+    "    program.append(cirq.measure(alice, key='b2'))\n",
+    "    # 5. Apply feed-forward corrections to bob\n",
+    "    program.append(cirq.Z.on(bob).with_classical_controls('b1'))\n",
+    "    program.append(cirq.X.on(bob).with_classical_controls('b2'))\n",
+    "    program.append(cirq.measure(bob, key='result'))\n",
+    "    return program"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "54968cd9",
+   "metadata": {},
+   "source": [
+    "## 4. Inspect the Circuit with Custom Gates"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d8ce308",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "circuit = cirq.Circuit(simple_teleportation())\n",
+    "print('Circuit with custom gates:')\n",
+    "print(circuit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8c98e44a",
+   "metadata": {},
+   "source": [
+    "## 5. Full Decomposition\n",
+    "We can ask Cirq to decompose all gates to primitive operations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "068d7430",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "decomposed_full = cirq.Circuit(cirq.decompose(circuit))\n",
+    "print('Fully decomposed:')\n",
+    "print(decomposed_full)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c9e98407",
+   "metadata": {},
+   "source": [
+    "## 6. Selective Decomposition\n",
+    "\n",
+    "The `keep` predicate controls where recursive decomposition stops:\n",
+    "- If `keep(op)` returns True, the operation is left as-is.\n",
+    "- If it returns False, Cirq tries to expand the operation via its decomposition methods.\n",
+    "\n",
+    "In the example below, we decompose only the custom gates (`BellPairGate`, `MessagePrepGate`) by returning False for them and True for everything else."
```

Comment:
Suggest mentioning that the diagram looks like the diagram from the teleportation notebook without custom gates.

Reply:
With only custom gates decompose it looks exactly the same as far as I can see. I'm talking about section 6.

Reply:
Ah you are right, I was looking at section 5. I've added a note about it being the same circuit as in teleportation.